lib/omniauth/login_dot_gov/authorization.rb

```diff
@@ -26,34 +27,68 @@ def redirect_url_params
           state: state,
           nonce: nonce,
         }
+
+        if aal
+          params[:aal] = aal
+        end
+
+        params
       end
 
-      def acr_values
-        if client.ial == 2
-          'http://idmanagement.gov/ns/assurance/loa/3'
+      def aal
+        if client.aal.nil?
+          nil
+        elsif VALID_AAL_VALUES.include?(client.aal.to_s)
+          client.aal
         else
-          'http://idmanagement.gov/ns/assurance/loa/1'
+          raise "Invalid AAL, choose one of #{VALID_AAL_VALUES}"
+        end
+      end
+
+      def acr_values
+        values = []
+
+        values << case client.ial
+        when 1, nil
+          'http://idmanagement.gov/ns/assurance/ial/1'
+        when 2
+          'http://idmanagement.gov/ns/assurance/ial/2'
+        when '2-strict'
+          'http://idmanagement.gov/ns/assurance/ial/2?strict=true'
+        when 0
+          'http://idmanagement.gov/ns/assurance/ial/0'
+        end
+
+        values << case aal
+        when 2
```

Comment:
**suggestion:** I think these two cases need to be strings, not integers

Reply:
I meant to make them integers (updated to drop the `to_s` above)

Reply:
Never mind, switched to strings